NEWS.md

```diff
@@ -40,6 +40,7 @@
 * `string_boundary_linter()` recognizes regular expression calls like `grepl("^abc$", x)` that can be replaced by using `==` instead (#1613, @MichaelChirico).
 * `unreachable_code_linter()` has an argument `allow_comment_regex` for customizing which "terminal" comments to exclude (#2327, @MichaelChirico). `# nolint end` comments are always excluded, as are {covr} exclusions (e.g. `# nocov end`) by default.
 * `format()` and `print()` methods for `lint` and `lints` classes get a new option `width` to control the printing width of lint messages (#1884, @MichaelChirico). The default is controlled by a new option `lintr.format_width`; if unset, no wrapping occurs (matching earlier behavior).
+* The `source_expression` passed to linters gains a fast way to query function call nodes using `source_expression$xml_find_function_calls()`. Use this to speed up linters using XPaths that start with `//SYMBOL_FUNCTION_CALL`. (@AshesITR, #2357)
```

Comment:
no mention of the new exported function?

Reply:
the NEWS should mention "See creating linters vignette for more details". It might help to have an illustrative benchmark as well.

Reply:
I would rework the NEWS bullet generally, it would be good to share some motivation for the change as it's pretty significant! We re-worked a large number of our exported linters to use this approach -- it's a bit of a paradigm shift for how downstreams should use {lintr} to write their own linters/XPaths. So motivating it well is important to drive adoption.

To me there's two big things (1) `//SYMBOL_FUNCTION_CALL` is easily the most common XPath component we see (2) It does not scale well for finding a medium-to-large number of functions since the `or` operator is not optimized in `libxml2`, as we saw for `backports_linter()`. Those insights make this new functionality a natural direction for new development, rather than us just releasing new features for the fun of it :)

Reply:
better?

Reply:
(threading got lost, alas, continued: https://github.com/r-lib/lintr/pull/2357#discussion_r1424908130. resolving this one)